pest/src/iterators/line_index.rs

```diff
@@ -7,38 +7,55 @@
 //! Inspired by rust-analyzer's `LineIndex`:
 //! <https://github.com/rust-lang/rust/blob/1.67.0/src/tools/rust-analyzer/crates/ide-db/src/line_index.rs>
 use alloc::vec::Vec;
+use core::cell::RefCell;
 
 #[derive(Clone)]
-pub struct LineIndex {
+pub struct LineIndex<'a> {
+    input: &'a str,
     /// Offset (bytes) the the beginning of each line, zero-based
-    line_offsets: Vec<usize>,
+    line_offsets: RefCell<Option<Vec<usize>>>,
```

Comment:
To preserve those UnwindSafe auto-traits, this would need to be Mutex or something like that instead of RefCell... but not sure if the performance improvement is still there in that case